logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/handler/legalhold/LegalHoldHandler.kt

```diff
@@ -71,12 +107,28 @@ internal class LegalHoldHandlerImpl internal constructor(
             if (selfUserId == legalHoldDisabled.userId) { // notify only for self user
                 userConfigRepository.setLegalHoldChangeNotified(false)
             }
-            legalHoldSystemMessagesHandler.handleDisable(legalHoldDisabled.userId)
+            handleConversationsForUser(legalHoldDisabled.userId)
+            legalHoldSystemMessagesHandler.handleDisabledForUser(legalHoldDisabled.userId, DateTimeUtil.currentIsoDateTimeString())
         }
 
         return Either.Right(Unit)
     }
 
+    override suspend fun handleNewMessage(message: MessageUnpackResult.ApplicationMessage, live: Boolean): Either<CoreFailure, Unit> {
+        val isStatusChangedForConversation = when (message.content.legalHoldStatus) {
```

Comment:
You can assign inside when statement 
```
when(val legalHoldStatus = message.content.legalHoldStatus)
```
and then just pass this value to `handleForConversation`